AK/MemoryStream.h

```diff
@@ -18,7 +18,12 @@ namespace AK {
 /// using a single read/write head.
 class FixedMemoryStream : public SeekableStream {
 public:
-    explicit FixedMemoryStream(Bytes bytes, bool writing_enabled = true);
+    enum class OpenMode {
```

Comment:
:bikeshed:, but `OpenMode` feels like the wrong term for this, it only made sense for `MappedFile` since we are actually opening/mapping something there. Maybe it should be a separate enum, `Protection`?

Reply:
Do we really need two enums for that? Using `Mode` for both use cases might be okay, what do you think?

Reply:
That works as well, I guess, depending on whether we will extend the `enum` for `MappedFile` in the future.